### PR TITLE
Analytics cleanup

### PR DESCRIPTION
**Pull request by frett**

### CODE REVIEW

File: library/xml-model/src/main/java/org/cru/godtools/xml/model/AnalyticsEvent.kt [hunk 1]
```diff
@@ -47,15 +50,20 @@ class AnalyticsEvent {
...
+    constructor(
+        action: String? = null,
+        delay: Int = 0,
+        systems: Set<AnalyticsSystem>? = null,
+        attributes: Map<String, String>? = null
+    ) {
+        this.action = action
         this.delay = delay
-        systems = emptySet()
+        this.systems = systems.orEmpty()
```
Comment (TheNoodleMoose) [inline library/xml-model/src/main/java/org/cru/godtools/xml/model/AnalyticsEvent.kt:61]: This just returns an empty string if there is no other string value right?

File: library/xml-model/src/main/java/org/cru/godtools/xml/model/AnalyticsEvent.kt [hunk 2]
```diff
@@ -47,15 +50,20 @@ class AnalyticsEvent {
...
         trigger = Trigger.DEFAULT
-        attributes = emptyMap()
+        this.attributes = attributes.orEmpty()
     }
 
-    internal constructor(parser: XmlPullParser) {
+    internal constructor(parent: Base, parser: XmlPullParser) {
```
Comment (TheNoodleMoose) [inline library/xml-model/src/main/java/org/cru/godtools/xml/model/AnalyticsEvent.kt:66]: Ah, so we need the ```parent``` to know which tool the event is coming from?

**Review by TheNoodleMoose — APPROVED** [review]
Looks good! Thanks for cleaning this up 🙂

File: library/xml-model/src/main/java/org/cru/godtools/xml/model/AnalyticsEvent.kt (hunk 2 above)
Comment (frett) [reply library/xml-model/src/main/java/org/cru/godtools/xml/model/AnalyticsEvent.kt:66]: yeah, the parent gives us access to the parsed object hierarchy. The Manifest at the root of the hierarchy has the tool and locale

File: library/xml-model/src/main/java/org/cru/godtools/xml/model/AnalyticsEvent.kt (hunk 1 above)
Comment (frett) [reply library/xml-model/src/main/java/org/cru/godtools/xml/model/AnalyticsEvent.kt:61]: empty set because `systems` is a `Set<AnalyticsSystem>`

Comment (TheNoodleMoose) [reply library/xml-model/src/main/java/org/cru/godtools/xml/model/AnalyticsEvent.kt:61]: Aahh gotcha its just returning an empty value for whatever the type is. Got it!

**Pull request merged**